classes/class-p4-search.php

```diff
@@ -0,0 +1,244 @@
+		protected $search_query;
+		/** @var array $templates */
+		public $templates;
+		/** @var array $context */
+		public $context;
+		/** @var array|bool|null $posts */
+		protected $posts;
+
+		/**
+		 * P4_Search constructor.
+		 *
+		 * @param string     $search_query The searched term.
+		 * @param array      $templates An indexed array with template file names. The first to be found will be used.
+		 * @param array|null $context An associative array with all the context needed to render the template found first.
+		 */
+		public function __construct( $search_query, $templates = [ 'search.twig', 'archive.twig', 'index.twig' ], $context = null ) {
+			$this->search_query = $search_query;
+			$this->templates    = $templates;
+
+			if ( $context ) {
+				$this->context  = $context;
+			} else {
+				$this->context  = Timber::get_context();
+				$this->posts    = Timber::get_posts();
+				$this->set_context( $this->context );
+			}
+		}
+
+		/**
+		 * Sets the P4 Search page context.
+		 *
+		 * @param array $context Associative array with the data to be passed to the view.
+		 */
+		protected function set_context( &$context ) {
+			$this->add_general_context( $context );
+			$this->add_items_context( $context );
+			$this->add_filters_context( $context );
+		}
+
+		/**
+		 * Adds the general context for the Search page.
+		 *
+		 * @param array $context Associative array with the data to be passed to the view.
+		 */
+		protected function add_general_context( &$context ) {
+			global $wp_query;
+
+			/*
+			 * With no args passed to this call, Timber uses the main query which we filter for customisations via P4_Master_Site class.
+			 *
+			 * When customising this query, use filters on the main query to avoid bypassing SearchWP's handling of the query.
+			 */
+			$context['posts']        = $this->posts;
+			$context['search_query'] = $this->search_query;
+			$context['found_posts']  = $wp_query->found_posts;
+
+			$selected_sort = filter_input( INPUT_GET, 'orderby', FILTER_SANITIZE_STRING );
+			if ( ! in_array( $selected_sort, array_keys( $context['sort_options'] ), true ) ) {
+				$context['selected_sort'] = P4_Master_Site::DEFAULT_SORT;
+			} else {
+				$context['selected_sort'] = $selected_sort;
+			}
+
+			// Footer Items.
+			$context['footer_social_menu']    = wp_get_nav_menu_items( 'Footer Social' );
+			$context['footer_primary_menu']   = wp_get_nav_menu_items( 'Footer Primary' );
+			$context['footer_secondary_menu'] = wp_get_nav_menu_items( 'Footer Secondary' );
+			$context['copyright_text']        = get_option( 'copyright', '' );
+		}
+
+		/**
+		 * Adds the context for each Search result item.
+		 *
+		 * @param array $context Associative array with the data to be passed to the view.
+		 */
+		protected function add_items_context( &$context ) {
+
+			foreach ( $this->posts as $post ) {
+				switch ( $post->post_type ) {
+					case 'page':
+						if ( 'act' === basename( get_permalink( $post->post_parent ) ) ) {
```

Comment:
Note: in the future this will have to change according to https://jira.greenpeace.org/browse/PLANET-1411